src/lib/math/pcurves/pcurves_brainpool256r1/info.txt

```diff
@@ -0,0 +1,9 @@
+<defines>
+PCURVES_BRAINPOOL256R1 -> 20240608
+</defines>
+
+<module_info>
+name -> "PCurve brainpool256r1"
+brief -> "brainpool256r1"
+type -> "Internal"
```

Comment:
Not 100% sure, but `type -> "Internal"` hides the module from the application user. I.e. they would be barred from enabling/disabling it via `./configure.py`. 😞

Reply:
Seems to work

```
$  ./configure.py --disable-modules=pcurves_secp256k1
...
   INFO: Skipping (disabled by user): pcurves_secp256k1
...
```

Reply:
What doesn't work, however:

```
$  ./configure.py --minimized-build --enable-modules=pcurves_secp256k1
...
  ERROR: Module 'pcurves_secp256r1' is meant for internal use only
...
```

... that's inconsistent with the `--disable-modules` switch. 😓 Let's tackle that independently.

I remember that I [deliberately set up the internal modules that way](https://github.com/randombit/botan/pull/3032):

> `Internal` modules behave essentially like Public modules, but the library user cannot actively enable/disable them. They are supposed to contain code that shouldn't be interacted with directly by the library user. Frankly, I don't know of any such module apart from the recently introduced kyber_common.

... this was apparently changed later: https://github.com/randombit/botan/pull/3489 (for `--disable-modules`).

I don't have a hard objection for allowing the user to interact with internal modules, though, my initial model was that they aren't visible to the user altogether.

Reply:
> initial model was that they aren't visible to the user altogether.

The problem here is there are two different notions of internal we could use

* Something internal, which if enabled the user can't see (eg in a header) but which affects behavior (`pcurves_secp256r1` being enabled or not changes, for example, if hash2curve works for P-256)
* Something truly internal which only exists to satisfy some other dependency and otherwise does not have a visible change in behavior (if KMAC is not included, then the existence or absence of `cshake_xof` changes nothing, so there is no reason to allow the user to enable it)

Maybe the answer here is that none of these modules should be "Internal" since their absence *does* introduce user visible changes?